pkg/api/accesscontrol.go

```diff
@@ -308,14 +308,14 @@ func (hs *HTTPServer) declareFixedRoles() error {
 
 	dashboardsCreatorRole := ac.RoleRegistration{
 		Role: ac.RoleDTO{
-			Version:     1,
+			Version:     2,
 			Name:        "fixed:dashboards:creator",
 			DisplayName: "Dashboard creator",
 			Description: "Create dashboard in general folder.",
 			Group:       "Dashboards",
 			Permissions: []ac.Permission{
-				{Action: dashboards.ActionFoldersRead, Scope: dashboards.ScopeFoldersProvider.GetResourceScope("0")},
-				{Action: ac.ActionDashboardsCreate, Scope: dashboards.ScopeFoldersProvider.GetResourceScope("0")},
+				{Action: dashboards.ActionFoldersRead, Scope: dashboards.ScopeFoldersProvider.GetResourceScopeUID("general")},
```

Comment:
We need to figure out how to handle access for creation of dashboards without a folder ("General"). This worked fine when using scopes with id but when using uid we need to fake some kind of uid

Reply:
- Yes
- Yes

As a response to this we've agreed in blocking off the UID "general" from creation and provisioning

Reply:
Do you think it would make sense to change the UID "general" to something else if a customer already has a folder with UID "general"? Although that has its own risks of breaking customer workflows which we might not know about.

Another option is to note this restriction in our docs. @vtorosyan do we have any doc sections where we can make exclaimers about scenarios in which FGAC won't function as expected?

Reply:
I am fine with the decision, if this is the best option we have now. Sorry if this has been discussed already somewhere, but would be great to verify that blocking the creation of UID "general" does not have any side-effect elsewhere. 

Also to double check, users won't be able to use FGAC for only dashboards/folders right?

And yeah we definitely need to document this. Let's perhaps put it in the [main homepage](https://grafana.com/docs/grafana/latest/enterprise/access-control/) under "Caveats" and ask for a feedback from docs.

Reply:
Re side-effects - I can't think of any, but I'll ask in #grafana-backend about it, maybe others can think of something.

I'm not sure I understand your second question. though.

Reply:
I've created https://github.com/grafana/grafana-enterprise/issues/3090 to track the doc change.

Reply:
> Also to double check, users won't be able to use FGAC for only dashboards/folders right?

Users having a folder/dashboard with uid "general" would be unable to apply FGAC permissions over those folders/dashboards. Seems to be a very niche scenario

Reply:
By the way, it doesn't look like anyone is objecting to us restricting folder UID options: https://raintank-corp.slack.com/archives/CMSUYD6H3/p1648636992684959

So I think we're good to merge this, and can work on doc changes in a separate PR.